doc/reference/autoinstall-reference.rst

```diff
@@ -3,184 +3,1013 @@
+must be present (unless the :ref:`user-data section <ai-user-data>` is present,
+in which case it is optional).
+
+A mapping that can contain keys, all of which take string values:
+
+realname
+~~~~~~~~
+
+The real name for the user. This field is optional.
+
+username
+~~~~~~~~
+
+The user name to create.
+
+hostname
+~~~~~~~~
+
+The hostname for the system.
+
+password
+~~~~~~~~
+
+The password for the new user, encrypted. This is required for use with
+``sudo``, even if SSH access is configured.
+
+The crypted password string must conform to what
+`passwd <https://manpages.ubuntu.com/manpages/jammy/en/man1/passwd.1.html>`_
+expects. Depending on the special characters in the password hash, quoting may
+be required, so it's safest to just always include the quotes around the hash.
+
+Several tools can generate the crypted password, such as ``mkpasswd`` from the
+``whois`` package, or ``openssl passwd``.
+
+Example:
+
+.. code-block:: yaml
+
+    identity:
+      realname: 'Ubuntu User'
+      username: ubuntu
+      password: '$6$wdAcoXrU039hKYPd$508Qvbe7ObUnxoj15DRCkzC3qO7edjH0VV7BPNRDYK4QR8ofJaEEF2heacn0QgD.f8pO8SNp83XNdWG6tocBM1'
+      hostname: ubuntu
+
+.. _ai-active-directory:
+
+active-directory
+----------------
+
+* **type:** mapping, see below
+* **default:** no default
+* **can be interactive:** yes
+
+Accepts data required to join the target system in an Active Directory domain.
+
+A mapping that can contain keys, all of which take string values:
+
+admin-name
+~~~~~~~~~~
+
+A domain account name with privilege to perform the join operation. That
+account's password will be requested during runtime.
+
+domain-name
+~~~~~~~~~~~
+
+The Active Directory domain to join.
 
 .. _ai-ubuntu-pro:
 
 ubuntu-pro
-==========
+----------
+
+* **type:** mapping, see below
+* **default:** see below
+* **can be interactive:** yes
+
+token
+~~~~~
+
+* **type:** string
+* **default:** no token
+
+A contract token to attach to an existing Ubuntu Pro subscription.
+
+.. _ai-ssh:
+
+ssh
+---
+
+* **type:** mapping, see below
+* **default:** see below
+* **can be interactive:** yes
+
+Configure SSH for the installed system. A mapping that can contain keys:
+
+install-server
+~~~~~~~~~~~~~~
+
+* **type:** boolean
+* **default:** ``false``
 
-stub
+Whether to install OpenSSH server in the target system.
+
+authorized-keys
+~~~~~~~~~~~~~~~
+
+* **type:** list of strings
+* **default:** ``[]``
+
+A list of SSH public keys to install in the initial user's account.
+
+allow-pw
+~~~~~~~~
+
+* **type:** boolean
+* **default:** ``true`` if ``authorized_keys`` is empty, ``false`` otherwise
+
+.. _ai-codecs:
+
+codecs
+------
+
+* **type:** mapping, see below
+* **default:** see below
+* **can be interactive:** no
+
+Configure whether common restricted packages (including codecs) from
+[multiverse] should be installed.
+
+install
+~~~~~~~
+
+* **type:** boolean
+* **default:** ``false``
+
+Whether to install the ubuntu-restricted-addons package.
```

Comment:
Should we link to the package? (and/or, say what it's for)

Reply:
That sounds like a great idea but I'm not sure what that would be!  The best I've come up with is the description in the control file but that's not much better:
https://git.launchpad.net/ubuntu/+source/ubuntu-restricted-addons/tree/debian/control#n13

@local-optimum - do you have a suggestion here for how we describe what's going on with this "codecs" functionality?